config/config.go

```diff
@@ -501,6 +521,15 @@ func (t *SyslogFacility) UnmarshalYAML(u func(interface{}, bool) error) (err err
 	return nil
 }
 
+func (t *HookEnum) UnmarshalYAML(u func(interface{}, bool) error) (err error) {
+	t.Ret, err = enumUnmarshal(u, map[string]interface{}{
+		"pre-and-post-command": &HookCommand{},
```

Comment:
I thought about this before giving it such a long name. I worry that if we give it a short, intuitive name like `command` that there will be sysadmin mistakes of putting a pre-hook in as a `command`. I've never used a hook system before in which a single command was run as both pre and post with only a change in environment. I like the elegance of the design you've suggested, but I'm concerned that sysadmins who don't read the docs carefully enough are likely to accidentally run their pre-hooks as post-hooks also if the name of the type isn't clear.

I agree it could be less clunky. Maybe `multi-command`? `all-command` sounds nice, but what if there are different edges in the future? (… thinking …) Maybe the right meaning here is expressed by `custom-command`, which hints to the administrator that she has to include some logic in the hook process for things to work properly. That also leaves the door open to edges other than `pre` and `post` in the future. All *custom* scripts should be implemented with the `if pre; elif post; else unsupported error` pattern so they break in an understandable way if applied to a new version with new edge semantics.

Reply:
I like the approach that the dehydrated Let's Encrypt client is taking:
https://github.com/lukas2511/dehydrated/blob/master/CHANGELOG#L53

Also Ctrl-F hook through the changelog, and have a look at the example hook https://github.com/lukas2511/dehydrated/blob/master/docs/examples/hook.sh

Bottom line: one hook.sh for all kinds of events. People seem to deal with it fairly well given the example hook as a starting point.